### ISSUE TITLE
Potential for handling payments

### ISSUE THREAD

**Issue by p2prider** [p2prider/p2prider#5]
Some good points raised in https://np.reddit.com/r/bangalore/comments/ths6tf/has_uberola_gotten_basically_unusable_or_is_it/i1aa3cc/

> [habitual_operation](https://old.reddit.com/user/habitual_operation) 1 point 4 minutes ago
> Personally, I think the idea is great. But I cannot get myself to agree with the part about payment. Ola and Uber being the middlemen makes haggling much less of a task. Second, the drivers cannot demand more than the shown amount at the end of the ride (or rather, we can refuse to pay, stating that we only pay the amount the app showed). Auto guys are notorious that way—they agree to drop us from one point to another at a certain price, then after getting to the destination, they suddenly demand a higher payment. When we refuse, more auto guys join in favour of the auto guy.
> 
> That payment part needs to be handled.
> 
> Ola and Uber track the ride throughout and know which route was taken. If the driver takes a wrong route, Ola or Uber absorb the additional cost (or probably pass it on to the auto guy—in either case, the customer doesn’t have to pay extra).

I personally think that for an MVP, we can make do without handling payments, but if the app grows and we get traction and more developers willing to contribute, we can start a minimal payments handling system on top of UPI.

**Comment by papasierragolf** [comment]
Yeah now everyone has phonepe, paytm, googlepay. Whatever drivers I noticed in the past week. So we can ask them to show QR code / or register their upi ID on the app and payment goes directly to them after a day in escrow? Where we can also generate some revenue from the interest
